src/Algebra/Properties/CommutativeSemigroup.agda

```diff
@@ -114,3 +114,14 @@ xy∙z≈yz∙x x y z =  trans (xy∙z≈y∙zx x y z) (sym (assoc y z x))
 
 xy∙z≈zx∙y :  ∀ x y z → (x ∙ y) ∙ z ≈ (z ∙ x) ∙ y
 xy∙z≈zx∙y x y z =  trans (xy∙z≈z∙xy x y z) (sym (assoc z x y))
+
+------------------------------------------------------------------------------
+-- Some permutation laws for _∙_ for four factors.
+------------------------------------------------------------------------------
+
+xy∙zu≈xz∙yu :  ∀ x y z u → (x ∙ y) ∙ (z ∙ u) ≈ (x ∙ z) ∙ (y ∙ u)
```

Comment:
I'm don't think we want to go listing all 120 permutation rules for 4 variables. Instead at this level of complexity we should be trying to use the library's monoid solvers instead.

Reply:
It is written "Some permutations". I set only few ones that I use in Nat.Binary.
If we need many of them, then we could try to use monoid solver.
OK?

Reply:
Again it's better to be consistent. Either we include all of them or none of them. Otherwise how can people predict which ones are present and which ones are missing?

Reply:
This leads us to a trap. 
Either to apply a monoid solver for several rather simple cases, with complicating the code with preparing parameters for the solver,
or to implement ``xy+zu≡xz+yu,  xy*zu≡xz*yu`` in Binary.Properties individually for Bin
(and then, probably, to do this in DivMod.agda, GCD.agda -- for CommutativeSemiring). 
??

Reply:
> Either to apply a monoid solver for several rather simple cases, with complicating the code with preparing parameters for the solver,

I agree, writing out for the solver doesn't always look nice. On the `experimental` branch we have a new monoid solver from @oisdk which doesn't require you to write out the full equation. This should be available after the next release of Agda.

In the meantime however I would recommend you using the existing solver. I don't think that your second proposal of proving it individually in multiple files is not the way to go.

Reply:
In e-mail, your last sentence ends with "is not the way to go",
which makes it a great difference. 
Probably, you have pressed "Comment", and the copy went to e-mail. 
And after this, you have corrected the message by "Edit".
Right?

With this, I conclude the following:
a message copy in e-mail often is not a fresh message on GitHub,
and it is better to read GitHub after seeing the corresponding message in e-mail.
??

Reply:
> In the meantime however I would recommend you using the existing solver

Or to target your PRs at `experimental` so that you may already start
using the new solver! It will make a great test case for it too.

Reply:
Let us delay this and wait until it becomes _master_.
Even with _master_ I have too many things to change.

I use the CommutativeMonoid solver in the coming Binary.Minus.